scripts/check_types.py fails on Windows: termios attr-defined errors + demands stubs not in the lockfile

﻿### Bug Description

`make type-check` / `python scripts/check_types.py` fails on Windows with 20 `attr-defined` errors that do not exist in CI:

```
livekit-agents\livekit\agents\cli\readchar.py:168: error: Module has no attribute "tcgetattr"  [attr-defined]
... (termios/tty attributes, 20 errors total)
```

and additionally demands stub packages that are not in the `typing` dependency group (`types-colorama`, `types-jsonschema`, `types-nanoid`, `types-pywin32`), instructing the contributor to `uv add` them — which would dirty `pyproject.toml`/`uv.lock` with Windows-only entries.

### Cause

mypy analyzes platform-conditional code for the *host* platform by default. CI runs the type gate on Ubuntu only, so the Unix branch of `cli/readchar.py` type-checks against a full `termios` stub there; on Windows the same branch resolves against the empty win32 stub, and the win32 branches additionally pull `colorama`/`pywin32` stubs that the lockfile (correctly) doesn't carry.

### Expected Behavior

`make type-check` produces the same result on every contributor OS as it does in CI.

### Proposed fix

Pass `--platform linux` to mypy in `scripts/check_types.py`. CI already runs on Linux, so its behavior is byte-identical; Windows/macOS contributors get CI parity. Verified locally on Windows 11 / Python 3.12: with the flag, a cold-cache run passes with zero extra stubs ("Success: no issues found in 625 source files"); without it, the 20 termios errors plus the missing-stub failure appear. A contributor who wants host-platform analysis can still override, since user-supplied mypy args are appended after the default.

I have a PR ready for this.

--- a/scripts/check_types.py
+++ b/scripts/check_types.py
@@ -98,6 +98,15 @@ def main() -> None:
         *uv_run_args,
         "mypy",
         "--untyped-calls-exclude=smithy_aws_core",
+        # Analyze for linux regardless of the host OS, matching the CI gate.
+        # Host-platform analysis breaks on Windows: the unix branch of
+        # cli/readchar.py resolves against the empty win32 termios stub
+        # (20 attr-defined errors) and the win32 branches demand stubs the
+        # typing group intentionally doesn't carry (types-colorama,
+        # types-pywin32, ...). User-supplied args come later, so an explicit
+        # --platform still overrides this default.
+        "--platform",
+        "linux",
         *pkg_args,
         *mypy_args,
     ]

